.github/workflows/build-install-test-snap-lxc.yaml

```diff
@@ -0,0 +1,83 @@
+name: build-install-test-snap-lxc
+
+on:
+  workflow_call:
+    inputs:
+      lxc-image:
+        required: true
+        type: string
+      ubuntu-image:
+        required: false
+        type: string
+        default: "ubuntu-latest"
+      snapcraft-channel:
+        required: false
+        type: string
+        default: "latest/stable"
+      snap-name:
+        required: true
+        type: string
+      branch-name:
+        required: true
+        type: string
+      snapcraft-args:
+        required: false
+        type: string
+      snap-install-args:
+        required: false
+        type: string
+      enable-experimental-extensions-env:
+        required: false
+        type: boolean
+      test-script:
+        description: "Custom script to run some custom snap testing"
+        required: false
+        type: string
+
+jobs:
+  build:
+    uses: ubuntu-robotics/snap_workflows/.github/workflows/build-install-test-snap.yaml@main
+    with:
+      ubuntu-image: ${{ inputs.ubuntu-image }}
+      snapcraft-channel: ${{ inputs.snapcraft-channel }}
+      branch-name: ${{ inputs.branch-name }}
+      snap-name: ${{ inputs.snap-name }}
+      snap-install-args: ${{ inputs.snap-install-args }}
+
+  test-on-lxc:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/download-artifact@v3
+      name: ${{ inputs.snap-name }}-${{ inputs.branch-name }}
+      with:
+        path: .
+    - name: Setup LXD
+      uses: canonical/setup-lxd@v0.1.1
+      with:
+        channel: latest/stable
+    - name: Launch container
+      run: |
+        lxc launch '${{ inputs.lxc-image }}' lxc-container
+        sleep 10 ## TODO(investigate): without sleep the next steps are flaky
+    - name: Push snap to lxc container
+      run: |
+        cd ${{ inputs.snap-name }}-${{ inputs.branch-name }}
+        lxc file push ${{ inputs.snap-name }}_*.snap lxc-container/root/
+    - name: Install snap with flags
+      if: inputs.snap-install-args != ''
```

Comment:
Maybe you can make those args an environment variable and pass it to the command. Thus avoiding this 'if/else'.

Reply:
Something along the lines of

```yaml
- name: Install snap with flags
  env:
    SNAP_INSTALL_ARGS: inputs.snap-install-args
  run: |
        lxc list
        lxc exec lxc-container -- sh -c "sudo snap install ${{ inputs.snap-name }}_*.snap '${SNAP_INSTALL_ARGS}'"
```

Reply:
ye I have the same in the other template, is it ok if I address those in the refactor ticket?